cider/exec/plan/validator/CiderPlanValidator.cpp

```diff
@@ -48,8 +48,15 @@ PlanSlice CiderPlanValidator::constructPlanSlice(
 
 bool CiderPlanValidator::isSupportedSlice(
     const PlanSlice& plan_slice,
+    const substrait::Plan& plan,
     const generator::FrontendEngine& from_platform) {
-  // TODO: (yma11) add function look up and rule based check
+  try {
+    auto c_plan = constructSubstraitPlan(plan_slice, plan);
+    auto translator = std::make_shared<generator::SubstraitToRelAlgExecutionUnit>(c_plan);
```

Comment:
can we create a new translator based on `PlanSlice`?
I feel `substraitPlan` -> `PlanSlice` -> `SubstraitPlan` -> `translate` is redundancy

Reply:
Even we do it by providing a new translator for PlanSlice, the original substraitPlan is still needed as it contains some needed info that PlanSlice doesn't hold. The logic is that we need construct a new plan based on the PlanSlice and then do compile using existing API.